### DISCUSSION TITLE
如何在模板中获取到当前页面url？

### DISCUSSION THREAD

**Discussion by WAGFS** [halo-dev/halo#6824] [Q&A] [answered]
在对主题做一些seo优化的时候发现，在Thymeleaf 模板中想要获取到当前页面url如果不使用`javascript`操作是一件比较麻烦的事，翻阅Thymeleaf 文档的时候发现之前有提供`#httpServletRequest`这类的对象，通过调用`#httpServletRequest.requestURL`可以拿到当前页面url，但现在不太行了。请问现在有什么api可以直接拿到当前页面url吗？
![image](https://github.com/user-attachments/assets/7c0959c8-38a1-4a54-84e2-8691b0319265)
这里还有几个其他的问题想要顺便问一下：
1. 有没有什么配置可以让模板变量中返回的url为绝对路径？例如在使用`post.spec.cover`时，如果是后台上传的会返回相对路径，这让我在处理页面的结构化数据时需要加一些额外的判断
2. 文章页有没有模板变量可以返回纯文字的文章内容，在使用`post.content.content`和`post.content.raw`时看起来两者返回的内容并没有明显差别，都是一段`html`，目前暂时是通过`javascript`来进行处理
![image](https://github.com/user-attachments/assets/6ecb21e3-9ea6-4053-bf23-c6486483982b)

文章页结构化数据格式：
![image](https://github.com/user-attachments/assets/0bd6f848-5f87-4ad6-ae85-6799b3966d74)

**Comment by guqing** [comment, marked as answer]
> 1. 在Thymeleaf 模板中想要获取到当前页面url

`${#ctx.getExchange().getRequest().getPathWithinApplication()}`

> 2. 有没有什么配置可以让模板变量中返回的url为绝对路径

目前没有提供这样的方法，只能通过 js 实现，可以使用 https://alpinejs.dev/ 这样的库来配合写模板可能是一个好的办法

> 3. 文章页有没有模板变量可以返回纯文字的文章内容

没有，参考第 2 点，示例项目 https://github.com/halo-dev/theme-earth/blob/main/src/alpine-data/share.ts

**Reply by WAGFS** [reply]
好的，谢谢~

**Reply by holwell** [reply]
> ${#ctx.getExchange().getRequest().getPathWithinApplication()}

谢谢
